utils/propertiesMatcher.ts

```diff
@@ -0,0 +1,252 @@
+  if (split.length < 1) return null;
+  if (split.length === 1) return [split[0], split[0]];
+  return [split[0], split[1]];
+}
+
+/**
+ *
+ * @param generate.charset Pattern like "a-z" or "a-z,1-5"
+ * @param generate.len Length to make random variable
+ * @param param1
+ * @returns
+ */
+export function generateDefault(generate: { charset: string; len: number }, { random = () => Math.random() } = {}) {
+  const validCharSets: number[][] = generate.charset.split(",").map(charRange).filter(Array.isArray);
+  if (validCharSets.length === 0) throw new Error("Expecing that we have a valid charset");
+  const max = validCharSets.reduce((acc, x) => x.reduce((x, y) => Math.max(x, y), acc), 0);
+  let i = 0;
+  const answer: string[] = Array(generate.len);
+  while (i < generate.len) {
+    const nextValue = Math.round(random() * max);
+    const inRange = validCharSets.reduce(
+      (acc, [lower, upper]) => acc || (nextValue >= lower && nextValue <= upper),
+      false
+    );
+    if (!inRange) continue;
+    answer[i] = String.fromCharCode(nextValue);
+    i++;
+  }
+  return answer.join("");
+}
+
+function withPattern<A>(value: unknown) {
+  if (matchPattern.test(value)) return matches.regex(RegExp(value.pattern));
+  return matches.string;
+}
+export function matchNumberWithRange(range: string) {
+  const matched = rangeRegex.exec(range);
+  if (!matched) return matches.number;
+  const [, left, leftValue, , rightValue, , right] = matched;
+  return matches.number
+    .validate(
+      leftValue === "*" ? (_) => true : left === "[" ? (x) => x >= Number(leftValue) : (x) => x > Number(leftValue),
+      leftValue === "*" ? "any" : left === "[" ? `greaterThanOrEqualTo${leftValue}` : `greaterThan${leftValue}`
+    )
+    .validate(
+      rightValue === "*" ? (_) => true : right === "]" ? (x) => x <= Number(rightValue) : (x) => x < Number(rightValue),
+      rightValue === "*" ? "any" : right === "]" ? `lessThanOrEqualTo${rightValue}` : `lessThan${rightValue}`
+    );
+}
+function withIntegral(parser: matches.Parser<unknown, number>, value: unknown) {
+  if (matchIntegral.test(value)) {
+    return parser.validate(Number.isInteger, "isIntegral");
+  }
+  return parser;
+}
+function withRange(value: unknown) {
+  if (matchRange.test(value)) {
+    return matchNumberWithRange(value.range);
+  }
+  return matches.number;
+}
+const isGenerator = matches.shape({ charset: matches.string, len: matches.number }).test;
+function defaultNullable<A>(parser: matches.Parser<unknown, A>, value: unknown) {
+  if (matchDefault.test(value)) {
+    if (isGenerator(value.default)) return parser.defaultTo(parser.unsafeCast(generateDefault(value.default)));
+    return parser.defaultTo(parser.unsafeCast(value.default));
+  }
+  if (matchNullable.test(value)) return parser.optional();
+  return parser;
+}
+export function guardAll<A>(value: A): matches.Parser<unknown, GuardAll<A>> {
+  if (!isType.test(value)) {
+    // deno-lint-ignore no-explicit-any
```

Comment:
:notebook: All of the any's are valid because of all the testing. The point of the any is, trust us devs. Well, the trust comes from the testing.